device/virtual/x86_64-kvm_x86_64-r0/virtual_chassis/virtual_chassis.py

```diff
@@ -0,0 +1,293 @@
+            for intf, br in self.brReq[instDir]:
+               hostIntf = hostName + "." + intf
+               data["neighbor_connections"][hostIntf] = br
+            if 'inband_address' in defCfg.keys():
+               iaddr = defCfg['inband_address']
+               ifname = self.ns + "veth" + instDir
+               ifpair = "inband"
+               data["inband_intf"] = ifname
+               data["inband_intf_pair"] = ifpair
+               data["inband_address"] = iaddr
+               runCmd(self.nss + " ip link add " + ifname + " type veth peer name "
+                       + ifpair)
+               runCmd(self.nss + " ip link set " + ifpair + " netns " + pid)
+               runCmd(self.nss + " ip link set dev " + ifname + " up")
+               runCmd(self.nss + " brctl addif br4chs " + ifname)
+               res = self.runCmdOnHost(hostName, " cat /sys/class/net/" +
+                                        ifpair + "/operstate")
+               while res.strip() != 'up':
+                  self.runCmdOnHost(hostName, " ip link set dev " + ifpair + " up")
+                  time.sleep(1)
+                  res = self.runCmdOnHost(hostName, " cat /sys/class/net/" +
+                                           ifpair + "/operstate")
+
+            vct_data = cwd + "/" + instDir + "/vct_connections.json"
+            with open (vct_data, 'w') as outfile:
+               json.dump(data, outfile)
+
+         else:
+            runCmd(" docker container stop " + hostName)
+            runCmd(" docker container rm " + hostName)
+      return
+
+   def addVethPairNs(self, intf, inst):
+      nss = self.nss
+      instName, pid = self.getInstInfo(inst)
+      ifn = instName[:9] + "." + intf
+      ifpair = intf
+
+      runCmd(nss + " ip link add " + ifn + " type veth peer name " + ifpair)
+      runCmd(nss + " ip link set " + ifn + " netns " + self.ns)
+      runCmd(nss + " ip link set " + ifpair + " netns " + pid)
+      runCmd(nss + " ip link set dev " + ifn + " up ")
+      self.runCmdOnHost(instName, " ip link set dev " + ifpair + " up")
+      return
+
+   def addToBr(self, intf, inst, br):
+      instName, _ = self.getInstInfo(inst)
+      ifn = instName + "." + intf
+      print "add %s to bridge %s " % (ifn, br)
+      runCmd(self.nss + " brctl addif " + br + " " + ifn)
+
+   def handleNeighConn(self, nck, ncv):
+      print "updateNeighbor %s %s" % (nck, self.oper)
+      br = "br" + nck
+      if self.oper == "create":
+         self.handleBridge(br)
+         for inst in ncv:
+            for srv in range(1, 33):
+               ifn = "eth" + str(srv)
+               if (ifn, br) not in self.readyIntfsInInst[inst]:
+                  if (ifn, br) in self.brReq[inst]:
+                     self.addToBr(ifn, inst, br)
+                  self.readyIntfsInInst[inst].append((ifn, br))
+      else:
+         self.handleBridge(br)
+      return
+
+   def verifyVirtualChassis(self, virtualTopology):
```

Comment:
Similar feedback as above, I would consider breaking this up into smaller helper methods based on the logical steps (e.g. check the connectivity, check the db, etc.) to make it easier to follow.

Reply:
ack